oci/layout/oci_transport.go

```diff
@@ -182,8 +209,14 @@ func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
 	if err != nil {
 		return imgspecv1.Descriptor{}, err
 	}
-
 	var d *imgspecv1.Descriptor
+	if ref.sourceIndex != -1 {
+		if len(index.Manifests) > ref.sourceIndex {
```

Comment:
Non-blocking? `ref.sourceIndex < len(index.Manifests)` feels more natural, at least it is consistent with `ref.sourceIndex != -1` (which is not `-1 != ref.sourceIndex`).